### PR TITLE
Fixing the Locked issue

### LINKED ISSUE #55: Improve the flatten script to save the hash and the date of any flattened contract

### PR DESCRIPTION
**Pull request by sullof**
In the first version of the contracts there was a typo. The events Locked and Unlocked have the parameter `tokenId` spelled as `tokendId`.
It is not possible to rename the parameter without losing the possibility of reading the old parameters which are necessary to get exact info about the date when the locked status changed.
The solution is to have three events:

`Locked(uint256 tokendId)`
`Unlocked(uint256 tokendId)`

and 

`Locked(uint256 tokenId, bool locked)`

The latter will be used from now going on when a token is locked or unlocked.
To solve the compatibility issue, we must forcefully emits Locked events to cover all the missing old events, so a new intermediate contract, EventPatch, has been add to expose two functions that allow to do so.

No need to patch the bridged versions because they have not been deployed yet.

### CODE REVIEW

**Comment by sullof** [comment]
We may remove the old events. In the end they are necessary only if there are complains. In that case we may easily build an app that retrieves the previous events from the blockchain and shows that the data are fair.
Opinions?

File: contracts/tokens/EventPatch.sol [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  uint256 public lastCheckedId;
+
+  function emitDefaultLockedEvent() public onlyOwner {
+    if (!defaultLockedEmitted) {
+      emit DefaultLocked(true);
+      defaultLockedEmitted = true;
+    }
+  }
+
+  function emitNewLockedEvent() public {
```
Comment (jerrybassat7) [inline contracts/tokens/EventPatch.sol]: Questions About this Function.  We want it to be only Public?  Anyone can call a new locked event?    Also dont we handle the case where we unlock event something like ; emit Locked(i, false);

**Review by jerrybassat7 — APPROVED** [review]
I think it makes sense to Remove the Old events since they are no more used. If we need them for something we can manually add them to the abi we use for what we need?

File: contracts/interfaces/IERC721DefaultLockable.sol [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.0;
+
+interface IERC721DefaultLockable {
+  // Must be emitted one time, when the contract is deployed,
+  // defining the default status of any token that will be minted
+  event DefaultLocked(bool locked);
+
+  // Must be emitted any time the status changes
+  event Locked(uint256 indexed tokenId, bool locked);
```
Comment (jerrybassat7) [inline contracts/interfaces/IERC721DefaultLockable.sol:10]: How does "indexed" work here?

**Pull request merged**